### PR TITLE
SC-16308, SC-16307 add `hostNetwork` page and pending issue info

### PR DESCRIPTION
**Pull request by boratanrikulu**

### CODE REVIEW

**Review by yagogou — APPROVED** [review]
LGTM

**Comment by yagogou** [comment]
Pinging @michaelhigham, so he is aware of this addition

File: docs/agents/sematext-agent/kubernetes/hostnetwork.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+title: `hostNetwork` Setting Usage for Control Plane Components Monitoring
+description: Explains why we need `hostNetwork` access and how to turn it off if desired.
+
+## What is `hostNetwork`?
+
+In Kubernetes, the `hostNetwork` is a configuration option that allows a container to share the network namespace of its host node. When `hostNetwork` is enabled for a pod, it means that the pod's containers will use the networking stack of the underlying host machine instead of having their own isolated network stack.
+
+## Sematext Agent and `hostNetwork` Access
+
+Sematext Agent requires `hostNetwork` access to monitor the control plane components. By default, the `hostNetwork` setting is set to `true` starting from version 3.0.0 of Sematext Agent.
```
Comment (cpipilas) [inline docs/agents/sematext-agent/kubernetes/hostnetwork.md:10]: @boratanrikulu `hostnetwork` access or enabled?

File: docs/monitoring/spm-faq.md [hunk 2]
```diff
@@ -731,6 +731,28 @@ After that, you should be able to add the key to your system.
...
 
+### How to prevent "Pending" status when installing Sematext Agent on a Kubernetes Cluster?
+
+If you notice that the pods for Sematext Agent are stuck in a "Pending" status with an error similar to "Port is already in use" during the installation on your Kubernetes Cluster, it may related to `hostNetwork` setting. 
+
+Sematext Agent requires `hostNetwork` access in order to access and monitor the Kubernetes control plane components. However, if you enable `hostNetwork` and the default port (8675) is already being used on your host, it can result in the Sematext Agent pods remaining in a "Pending" state. To resolve this problem, you need to modify the port being used.
+
+If you have concerns about `hostNetwork` usage or you don't want to use it in your system, please refer to [this page](https://sematext.com/docs/agents/sematext-agent/kubernetes/hostnetwork/).
+
+If you will continue to use `hostNetwork` (which is required for fully featured Kubernetes monitoring), please follow the steps below:
```
Comment (cpipilas) [inline docs/monitoring/spm-faq.md:742]: ...please follow the steps below, depending on the installation method you are using: ....

File: docs/monitoring/spm-faq.md [hunk 3]
```diff
@@ -731,6 +731,28 @@ After that, you should be able to add the key to your system.
...
+
+Here are the instructions based on the installation method you are using:
```
Comment (cpipilas) [inline docs/monitoring/spm-faq.md:744]: remove this

File: docs/agents/sematext-agent/kubernetes/hostnetwork.md (hunk 1 above)
Comment (boratanrikulu) [reply docs/agents/sematext-agent/kubernetes/hostnetwork.md:10]: Hmm, I think they're both OK. Do you think `enabled` would be better?

Comment (boratanrikulu) [reply docs/agents/sematext-agent/kubernetes/hostnetwork.md]: I updated it like this

```
## Sematext Agent and `hostNetwork` Access

Sematext Agent requires `hostNetwork: true` setting to monitor the control plane components. By default, the `hostNetwork` setting is set to `true` starting from version 3.0.0 of Sematext Agent.
```

Comment (cpipilas) [reply docs/agents/sematext-agent/kubernetes/hostnetwork.md]: Looks good now.

**Review by cpipilas — APPROVED** [review]
LGTM

File: docs/monitoring/spm-faq.md [hunk 4]
```diff
@@ -731,6 +731,26 @@ After that, you should be able to add the key to your system.
 wget -O - https://pub-repo.sematext.com/debian/sematext.gpg.key | sudo apt-key add -
 ```
 
+### How to prevent "Pending" status when installing Sematext Agent on a Kubernetes Cluster?
+
+If you notice that the pods for Sematext Agent are stuck in a "Pending" status with an error similar to "Port is already in use" during the installation on your Kubernetes Cluster, it may related to `hostNetwork` setting.
```
Comment (otisg) [inline docs/monitoring/spm-faq.md:736]: it may related ==> it might be related

File: docs/monitoring/spm-faq.md [hunk 5]
```diff
@@ -731,6 +731,26 @@ After that, you should be able to add the key to your system.
...
 
+### How to prevent "Pending" status when installing Sematext Agent on a Kubernetes Cluster?
+
+If you notice that the pods for Sematext Agent are stuck in a "Pending" status with an error similar to "Port is already in use" during the installation on your Kubernetes Cluster, it may related to `hostNetwork` setting. 
+
+Sematext Agent requires `hostNetwork` access in order to access and monitor the Kubernetes control plane components. However, if you enable `hostNetwork` and the default port (8675) is already being used on your host, it can result in the Sematext Agent pods remaining in a "Pending" state. To resolve this problem, you need to modify the port being used.
+
+If you have concerns about `hostNetwork` usage or you don't want to use it in your system, please refer to [this page](https://sematext.com/docs/agents/sematext-agent/kubernetes/hostnetwork/).
+
+If you will continue to use `hostNetwork` (which is required for fully featured Kubernetes monitoring), please follow the steps below, depending on the installation method you are using:
```
Comment (otisg) [inline docs/monitoring/spm-faq.md:742]: If you will continue to use ==> If you want to proceed with

File: docs/monitoring/spm-faq.md [hunk 6]
```diff
@@ -731,6 +731,26 @@ After that, you should be able to add the key to your system.
...
+
+**Helm:**
+
+Include the following parameter in your regular helm installation command: `--set agent.service.port=<Free-Port>`. Replace `<Free-Port>` with an available port number that is not already in use.
```
Comment (otisg) [inline docs/monitoring/spm-faq.md:746]: Replace `<Free-Port>` with an available port number that is not already in use. ==> Replace `<Free-Port>` with a port that is not in use.

File: docs/monitoring/spm-faq.md [hunk 7]
```diff
@@ -731,6 +731,26 @@ After that, you should be able to add the key to your system.
...
+
+**Kubectl:**
+
+Update the `API_SERVER_PORT` environment variable with the new `<Free-Port>` value in your YAML file. The new `<Free-Port>` value should be an available port number that is not already in use. Additionally, make sure to update the port values for `livenessProbe`, `startupProbe`, and `ports` accordingly, using the same `<Free-Port>` value.
```
Comment (otisg) [inline docs/monitoring/spm-faq.md:750]: The new `<Free-Port>` value should be an available port number that is not already in use ==>
The new `<Free-Port>` value should be a port that is not in use

**Pull request merged**